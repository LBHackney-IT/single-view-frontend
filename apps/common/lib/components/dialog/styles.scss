@import "~lbh-frontend/lbh/base";
@import "~lbh-frontend/lbh/components/lbh-dialog/_dialog.scss";

[data-reach-dialog-overlay] {
  z-index: 100;
}

.lbh-dialog__close {
  margin-top: 1.5rem;
  margin-right: 1.5rem;
}

.lbh-dialog__title {
  margin-right: 2rem;
}
