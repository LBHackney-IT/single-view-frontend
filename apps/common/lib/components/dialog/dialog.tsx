import React, { ComponentPropsWithoutRef, forwardRef } from "react";

import { Dialog as ReachDialog } from "@reach/dialog";
import cn from "classnames";
import "@reach/dialog/styles.css";

import { Heading } from "../heading";
import "./styles.scss";

export interface DialogProps extends ComponentPropsWithoutRef<"div"> {
  isOpen: boolean;
  onDismiss: () => void;
  title: string;
}

export const Dialog = forwardRef<HTMLDivElement, DialogProps>(function Dialog(
  { isOpen, onDismiss, children, className, title, ...props },
  ref,
) {
  return (
    <ReachDialog
      ref={ref}
      isOpen={isOpen}
      onDismiss={onDismiss}
      className={cn("lbh-dialog", className)}
      aria-label={title}
      {...props}
    >
      <Heading as="h2" variant="h2" className="lbh-dialog__title">
        {title}
      </Heading>
      <button type="button" onClick={onDismiss} className="lbh-dialog__close">
        <span className="govuk-visually-hidden">Close</span>

        <svg width="18" height="18" viewBox="0 0 13 13" fill="none">
          <path
            d="M-0.0501709 1.36379L1.36404 -0.050415L12.6778 11.2633L11.2635 12.6775L-0.0501709 1.36379Z"
            fill="#0B0C0C"
          />
          <path
            d="M11.2635 -0.050293L12.6778 1.36392L1.36404 12.6776L-0.0501709 11.2634L11.2635 -0.050293Z"
            fill="#0B0C0C"
          />
        </svg>
      </button>
      {children}
    </ReachDialog>
  );
});

export const DialogActions = forwardRef<HTMLDivElement, ComponentPropsWithoutRef<"div">>(
  function DialogActions({ className, ...props }, ref) {
    return <div ref={ref} className={cn("lbh-dialog__actions", className)} {...props} />;
  },
);
