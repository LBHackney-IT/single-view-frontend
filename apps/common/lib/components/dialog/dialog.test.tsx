import React, { FC, useState } from "react";

import { render, testA11y } from "@hackney/mtfh-test-utils";
import { screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import { Dialog, DialogActions } from "./dialog";

const Component: FC = ({ children }) => {
  const [isOpen, setOpen] = useState(false);
  return (
    <>
      <button type="button" onClick={() => setOpen(true)}>
        Toggle
      </button>
      <Dialog isOpen={isOpen} title="Dialog Title" onDismiss={() => setOpen(false)}>
        {children}
      </Dialog>
    </>
  );
};

test("it renders correctly", async () => {
  const { container } = render(
    <Component>
      <p>Content</p>
    </Component>,
  );
  expect(container).toMatchSnapshot();
  await testA11y(container);

  const toggle = screen.getByText("Toggle") as HTMLButtonElement;
  userEvent.click(toggle);

  expect(container).toMatchSnapshot();
  await testA11y(container);
});

test("it renders correctly with actions", async () => {
  const { container } = render(
    <Component>
      <p>Content</p>
      <DialogActions>
        <button type="button">Confirm</button>
        <a href="/">Cancel</a>
      </DialogActions>
    </Component>,
  );

  const toggle = screen.getByText("Toggle") as HTMLButtonElement;
  userEvent.click(toggle);

  expect(container).toMatchSnapshot();
  await testA11y(container);
});
