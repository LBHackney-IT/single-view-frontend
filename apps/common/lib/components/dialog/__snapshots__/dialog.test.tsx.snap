// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<div>
  <button
    type="button"
  >
    Toggle
  </button>
</div>
`;

exports[`it renders correctly 2`] = `
<div
  aria-hidden="true"
>
  <button
    type="button"
  >
    Toggle
  </button>
</div>
`;

exports[`it renders correctly with actions 1`] = `
<div
  aria-hidden="true"
>
  <button
    type="button"
  >
    Toggle
  </button>
</div>
`;
